### PR TITLE
fixed Wrange-loop-analysis warning with Apple clang-800.0.42.1

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

**Comment by arvidn** [comment]
it seems quite unnecessary to take a reference to an ``int32``.. it is a ``piece_index_t``, right? I suspect there's something wrong with this warning.

**Comment by aldenml** [comment]
yes, it looks strange to me, that's the reason I made the PR just with this, maybe something is missing in the type that hint the compiler it's not necessary.

**Comment by aldenml** [comment]
the warning is happening with the latest android NDK too (clang)

File: src/torrent.cpp [hunk 1]
```diff
@@ -1810,7 +1810,7 @@ namespace libtorrent
 				have_pieces.push_back(p.index);
 			}
 
-			for (auto const i : have_pieces)
```
Comment (arvidn) [inline src/torrent.cpp]: the patch against clang that added this warning suggests that removing ``const`` here would also silence the warning. I think that's preferable. It's unfortunate, though, that it seems to discriminate primitive and other types.

Comment (aldenml) [inline src/torrent.cpp]: interesting...well, removed `const&` and warning is gone

**Comment by arvidn** [comment]
lgtm

**Pull request merged**